mongoreplay/mongo_op_handler.go

```diff
@@ -312,20 +312,37 @@ func (bidi *bidi) handleStreamStateOutOfSync(stream *stream) {
 // stream's and tries to create whole protocol messages from them.
 func (bidi *bidi) streamOps() {
 	bidi.logvf(Info, "starting")
-	for {
+	ch0 := bidi.streams[0].reassembled
+	ch1 := bidi.streams[1].reassembled
+	for ch0 != nil || ch1 != nil {
 		var reassemblies []tcpassembly.Reassembly
 		var reassembliesStream int
 		var ok bool
 		select {
-		case reassemblies, ok = <-bidi.streams[0].reassembled:
+		case reassemblies, ok = <-ch0:
 			reassembliesStream = 0
-		case reassemblies, ok = <-bidi.streams[1].reassembled:
+			if !ok {
+				ch0 = nil
+			}
+		case reassemblies, ok = <-ch1:
 			reassembliesStream = 1
+			if !ok {
+				ch1 = nil
+			}
 		}
+		stream := bidi.streams[reassembliesStream]
 		if !ok {
-			break
+			if !stream.opTimeStamp.IsZero() {
```

Comment:
I think there should be just one EOF per bidirectional connection, so you can replace the break and move this EOF op generation to after the loop block.